am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -45,21 +52,36 @@ public void createResourceAccess(ExplicitAccessRecord explicitAccessRecord) {
     }
 
     /**
-     * Returns `resourceJson` when record with userId and resourceId exist and has READ permissions, otherwise null.
+     * Returns FilterResourceResponse when record with userId and resourceId exist and has READ permissions,
+     * otherwise null.
      *
      * @param userId       (accessorId)
      * @param resourceId   resource id
      * @param resourceJson json
      * @return resourceJson or null
+     * @throws UnsupportedPermissionsException when permissions are above 31 or below 0.
      */
-    public JsonNode filterResource(String userId, String resourceId, JsonNode resourceJson) {
+
+    public FilterResourceResponse filterResource(String userId, String resourceId, JsonNode resourceJson)
+        throws UnsupportedPermissionsException {
         AccessManagement explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
             dao -> dao.getExplicitAccess(userId, resourceId));
 
         if (explicitAccess == null) {
             return null;
         }
 
-        return Permissions.hasPermissionTo(explicitAccess.getPermissions(), Permissions.READ) ? resourceJson : null;
+        if (READ.isGranted(explicitAccess.getPermissions())) {
+            ConcurrentHashMap<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
```

Comment:
Please use `Map` to declare variable type. It helps coding agains interfaces and implementation.

Ps. you will still initiate variable as ConcurrentHashMap.

Reply:
👍